bolt-contracts/src/contracts/BoltChallenger.sol

```diff
@@ -194,101 +232,138 @@ contract BoltChallenger is IBoltChallenger {
+
         Challenge storage challenge = challenges[challengeID];
 
         if (challenge.status != ChallengeStatus.Open) {
             revert ChallengeAlreadyResolved();
         }
 
         if (challenge.openedAt + MAX_CHALLENGE_DURATION < Time.timestamp()) {
-            // If the challenge has expired without being resolved, it is considered lost.
-            challenge.status = ChallengeStatus.Lost;
+            // If the challenge has expired without being resolved, it is considered breached.
+            // It's cheaper to call `resolveExpiredChallenge()` directly in this case, but this
+            // case should still be handled here for consistency.
+            challenge.status = ChallengeStatus.Breached;
             _transferFullBond(challenge.challenger);
-            emit ChallengeLost(challengeID);
+            emit ChallengeBreached(challengeID);
             return;
         }
 
-        // Verify the validity of the header against the trusted block hash.
-        if (keccak256(proof.blockHeaderRLP) != trustedBlockHash) {
-            revert InvalidBlockHash();
+        // Check theintegrity of the proofs
+        if (
+            proof.txMerkleProofs.length != challenge.committedTxs.length
+                || proof.txIndexesInBlock.length != challenge.committedTxs.length
+        ) {
+            revert InvalidProofsLength();
         }
 
-        // Decode the RLP-encoded block header fields
-        BlockHeaderData memory blockHeader = _decodeBlockHeaderRLP(proof.blockHeaderRLP);
+        // Check the integrity of the trusted block hash
+        if (keccak256(proof.previousBlockHeaderRLP) != trustedPreviousBlockHash) {
+            revert InvalidBlockHash();
+        }
 
-        // Decode the committed raw signed transaction. Its sender will be the account to prove existence of.
-        TransactionDecoder.Transaction memory decodedTx = challenge.commitment.signedTx.decodeEnveloped();
+        // Decode the RLP-encoded block header of the previous block to the inclusion block.
+        //
+        // The previous block's state root is necessary to verify the account had the correct balance and
+        // nonce at the top of the inclusion block (before any transactions were applied).
+        BlockHeaderData memory previousBlockHeader = _decodeBlockHeaderRLP(proof.previousBlockHeaderRLP);
+
+        // Decode the RLP-encoded block header of the inclusion block.
+        //
+        // The inclusion block is necessary to extract the transaction root and verify the inclusion of the
+        // committed transactions. By checking against the previous block's parent hash we can ensure this
+        // is the correct block trusting a single block hash.
+        BlockHeaderData memory inclusionBlockHeader = _decodeBlockHeaderRLP(proof.inclusionBlockHeaderRLP);
+
+        // Check that the inclusion block is a child of the previous block
+        if (inclusionBlockHeader.parentHash != keccak256(proof.previousBlockHeaderRLP)) {
```

Comment:
Can you store `keccak256(proof.previousBlockHeaderRLP)` in a variable? Since this is also re-used above